src/targets/gpu/fuse_ck.cpp

```diff
@@ -161,10 +203,81 @@ struct find_ck_gemm
     }
 };
 
+static bool is_mul_module(const module& m)
+{
+    std::vector<std::string> result;
+    for(auto& ins : m)
+    {
+        if(starts_with(ins.name(), "@"))
+            continue;
+        if(contains({"multibroadcast", "contiguous"}, ins.name()))
+            continue;
+        if(ins.name() == "pointwise")
+        {
+            return is_mul_module(*ins.module_inputs().front());
+        }
+        else if(ins.name() == "mul")
+        {
+            return true;
```

Comment:
What if there is more operators after mul? It seems like it would be better to check if its returning mul and that the inputs to mul are input parameters. We could write a matcher for it:

```cpp
auto is_mul = match::arg(0)(match::name("mul")(match::all_of[match::inputs()](match::name("@param"))));
return match_instruction(m, std::prev(m.end()), is_mul);
```